src/Symfony/Component/Security/Core/Authorization/AccessDecisionManager.php

```diff
@@ -43,13 +49,9 @@ class AccessDecisionManager implements AccessDecisionManagerInterface
      */
     public function __construct(array $voters = array(), $strategy = self::STRATEGY_AFFIRMATIVE, $allowIfAllAbstainDecisions = false, $allowIfEqualGrantedDeniedDecisions = true)
     {
-        $strategyMethod = 'decide'.ucfirst($strategy);
-        if (!is_callable(array($this, $strategyMethod))) {
-            throw new \InvalidArgumentException(sprintf('The strategy "%s" is not supported.', $strategy));
-        }
-
+        $this->strategies = array();
```

Comment:
injecting an array of strategy objects does not make sense IMO: you only need a single strategy, and changing it after instantiation is forbidden.
You should just allow injecting a Strategy objct rather than a string, and handle other things in the configuration (allowing to specify either a builtin strategy, or a string, for BC reasons)